plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/price/frontend.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { getElement, store } from '@wordpress/interactivity';
+import '@woocommerce/stores/woocommerce/single-product';
+import type { SingleProductStore } from '@woocommerce/stores/woocommerce/single-product';
+import { sanitize } from 'dompurify'; // eslint-disable-line import/named
+
+// Stores are locked to prevent 3PD usage until the API is stable.
+const universalLock =
+	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
+
+const { state: singleProductState } = store< SingleProductStore >(
+	'woocommerce/single-product',
+	{},
+	{ lock: universalLock }
+);
+
+const ALLOWED_TAGS = [
+	'a',
+	'b',
+	'em',
+	'i',
+	'strong',
+	'p',
+	'br',
+	'span',
+	'bdi',
+	'del',
+	'ins',
+];
+const ALLOWED_ATTR = [
+	'class',
+	'target',
+	'href',
+	'rel',
+	'name',
+	'download',
+	'aria-hidden',
+];
+
+const productPriceStore = store(
+	'woocommerce/product-price',
+	{
+		state: {
+			get priceHTML(): string {
+				return singleProductState?.productData?.price_html || '';
+			},
+			get originalPriceHtml(): string {
+				return (
+					singleProductState?.originalProductData?.price_html || ''
+				);
+			},
+		},
+		callbacks: {
+			updatePrice: () => {
+				const element = getElement();
+
+				if ( ! element.ref ) {
+					return;
+				}
+
+				const { priceHTML, originalPriceHtml } =
+					productPriceStore.state;
```

Comment:
if we don't use this state in any directive, we can use it directly from the singleProductState without creating immediate derived state.

Reply:
Good call! Changed in https://github.com/woocommerce/woocommerce/pull/58384/commits/ff49f1c6ae7a0b152e81349f28cc5ba462f07001.